core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -202,7 +204,7 @@ public Map<String, SnapshotRef> refs() {
 
   @Override
   public UUID uuid() {
-    return UUID.randomUUID();
```

Comment:
Wow, I can't believe I wrote it like this originally :). Good fix, yes we do want a UUID but it should be consistent across the same Metadata Table. 

The only thing is if we extend this logic further is that the same base metadata table across different references won't return the same UUID. So for example "all manifests" for t1 may have a different UUID than "all manifests" for t1 in a different query.

I don't really see a way around that without persisting the UUID somewhere but considering metadata tables are really more logical tables, I think that's OK. If we wanted that strict definition for metadata tables, then we would have to throw an unsupported until we persisted those details somewhere.

For now though, I think this change is good as is.